host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -2120,7 +2120,8 @@ static MMAL_STATUS_T create_encoder_component(RASPIVID_STATE *state)
    if (state->encoding == MMAL_ENCODING_H264 &&
        state->slices > 1)
    {
-      int slice_row_mb = (state->height/16)/state->slices + (state->height/16)%state->slices?1:0; //round up
+      int frame_mb_rows = (state->height/16) + (state->height%16)?1:0;
+      int slice_row_mb = frame_mb_rows/state->slices + frame_mb_rows%state->slices?1:0; //round up
```

Comment:
Sorry, I still think that fails the desired criteria.
Height 128 and 2 slices sets slice_row_mb to 5, or 160 lines per slice. In fact many of the heights come out wrong. I'm just making up a simple spreadsheet that runs through the calculation for heights 128-180, prints the result, and then multiplies it back up. Heights 146-176 all end up with a slice_row_mb of 6 (192), when only over 160 requires 6.

I'll have a think about the logic myself - the modulo stuff feels very wrong, but I also appreciate that it may be required for handling 3 slices/frame.

Reply:
It's possible without the modulo.